### ISSUE TITLE
Possible crash on OSX 10.10?

### ISSUE THREAD

**Issue by jceipek** [mrRay/VVUVCKit#1]
I've been trying to use reacTIVision (https://github.com/mkalten/reacTIVision/), which seems to depend on the frameworks in this kit, and it crashes inside `VVUVCController.m`. Could that be a result of a bug in VVUVCKit? See https://github.com/mkalten/reacTIVision/issues/4

**Comment by mrRay** [comment]
howdy-

> Could that be a result of a bug in VVUVCKit?

sure!

1)- what happens if you comment out line 460 of `VVUVCController.m` and then rebuild/test?  does this prevent the crash?
2)- if (1) didn't prevent the crash, could you please let me know what kind of camera you're working with, and whether or not the test app in VVUVCKit exhibits the problem you're experiencing in reactivision?

cheers
: : ray

**Comment by jceipek** [comment]
It did prevent the crash, but then I get:

```
reacTIVision 1.5 (Apr  2 2015)

no DC1394 cameras found
1 AVFoundation camera found
2015-04-05 14:58:34.445 reacTIVision[6873:492154] -[VVUVCController initWithDeviceIDString:] ... CC242247UZLF6VVDU
2015-04-05 14:58:34.446 reacTIVision[6873:492154] -[VVUVCController initWithLocationID:] ... 0, 0
2015-04-05 14:58:34.479 reacTIVision[6873:492154] CameraControl Error: IOCreatePlugInInterfaceForService returned 0xe00002be.
camera: FaceTime HD Camera
format: 640x480, 30fps
render: opengl
```

I also have a PS3 webcam connected, but I can't get it to be recognized; I think something like https://github.com/inspirit/PS3EYEDriver would need to be integrated.

Running the test app without any changes, I get this:

```
2015-04-05 15:06:05.669 UVC Test App[6946:508457] Failed to connect (window) outlet from (AppDelegate) to (NSWindow): missing setter or instance variable
2015-04-05 15:06:09.516 UVC Test App[6946:508457] -[VVUVCController initWithDeviceIDString:] ... CC242247UZLF6VVDU
2015-04-05 15:06:09.516 UVC Test App[6946:508457] -[VVUVCController initWithLocationID:] ... 0, 0
2015-04-05 15:06:09.544 UVC Test App[6946:508457] CameraControl Error: IOCreatePlugInInterfaceForService returned 0xe00002be.
fish: Job 1, './UVC\ Test\ App ' terminated by signal SIGSEGV (Address boundary error)
```

Commenting out line 460 has the same result as with reactivision.

**Comment by mrRay** [comment]
> ”It did prevent the crash, but then I get:”

yep- you were getting the same error before, too.  basically, VVUVCKit encountered that error at some point, logged it (`CameraControl Error: IOCreatePlugInInterfaceForService returned 0xe00002be.`) and then tried to clean up and destroy the plugin interface.  what i asked you to do was comment out the line that tried to clean stuff up (inappropriately, it would appear!).  this may have prevented the crash- but the underlying error still stands: stuff’s not going to work for some reason.

i’d like to figure out how to reproduce this error so i can test against it locally.  you mentioned that your ps3 camera was plugged in- please unplug it, and then relaunch the test app.  does it crash or throw any errors?

**Comment by jceipek** [comment]
Whether I had the PS3 camera plugged in or not had no impact on the error: it happened in either case. Sorry for not specifying that before.

**Comment by mrRay** [comment]
okay, then what kind of camera are you having problems with?  it looked like you were using a built-in webcam, so if that’s the case…what kind of machine are you using?

**Comment by jceipek** [comment]
I was using the built-in FaceTime HD Camera (Apple Camera VendorID_0x106B ProductID_0x1570) on a MacBookPro11,3 (Retina, 15-inch, Late 2013) running Yosemite 10.10.2.
In case it matters, the machine has 16GB DDR3 1600 MHz memory, the processor is a 2.6 GHz Intel Core i7, and the graphics card combination is Intel Iris Pro/NVIDIA GeForce GT 750M 2048 MB.

**Comment by mrRay** [comment]
i don’t think this is working for you because this framework only controls UVC video devices, and as far as i can tell the built-in camera on your computer isn’t a UVC video device; i’m not even entirely sure if the camera is USB-based.  i have a friend with a late-2013 rmbp, and his built-in camera is only listed under “Hardware->Camera”, and it isn’t listed as a USB device at all.

i just pushed a commit to this framework that should resolve the crash you found when trying to clean up in case of an invalid address- thanks very much for bringing it to my attention!  as things stand now, if a UVC controller can’t be created, it releases itself and returns nil, which seems sensible and relatively safe.

did you ever get a driver for your ps3 webcam?  i’m curious if/how well that works…

**Comment by jceipek** [comment]
Ah, that makes sense.

Check out https://github.com/mkalten/reacTIVision/issues/5#issuecomment-95455809
